src/main/kotlin/com/sourcegraph/cody/agent/protocol/CurrentUserCodySubscription.kt

```diff
@@ -0,0 +1,27 @@
+package com.sourcegraph.cody.agent.protocol
+
+import com.google.gson.annotations.SerializedName
+import java.util.*
+
+data class CurrentUserCodySubscription(
+    val status: Status,
+    val plan: Plan,
+    val applyProRateLimits: Boolean,
+    val currentPeriodStartAt: Date,
+    val currentPeriodEndAt: Date,
+)
+
+enum class Plan {
```

Comment:
For me serialisation of enums was not working at least in older IJ versions.
Are you **sure** it works?
It was silently ignoring `@SerializedName` and serialising this to the order number of enum.

Reply:
It works for me. You can test it. I think the problem with `Speaker` is that we mix serialization with inheritance (on both - client and server sides)